### PR TITLE
[Merged by Bors] - feat(CategoryTheory/Bicategory/LocallyDiscrete): Make LocallyDiscrete a structure

### PR DESCRIPTION
**Pull request by callesonne**
Made `LocallyDiscrete` a structure, similar to `Discrete`, and added some basic API. `Coherence.lean` was also adjusted because of these changes. This should make it easier to move between a given type `C` and `LocallyDiscrete C`.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/CategoryTheory/Bicategory/Coherence.lean [hunk 1]
```diff
@@ -250,7 +250,7 @@ def inclusion (B : Type u) [Quiver.{v + 1} B] :
     Pseudofunctor (LocallyDiscrete (Paths B)) (FreeBicategory B) :=
   { -- All the conditions for 2-morphisms are trivial thanks to the coherence theorem!
     preinclusion B with
-    mapId := fun a : FreeBicategory B => Iso.refl (𝟙 a)
+    mapId := fun a => eqToIso <| by aesop_cat
```
Comment (joelriou) [inline Mathlib/CategoryTheory/Bicategory/Coherence.lean:253]:
```suggestion
    mapId := fun a => Iso.refl _
```

Otherwise, this looks good to me!

Comment (callesonne) [reply Mathlib/CategoryTheory/Bicategory/Coherence.lean]: Nice spot, thanks a lot!

**Comment by joelriou** [comment]
Could you move the description text of the PR above the line with the "---"? (Also, you may add the tag `awaiting-review` when you feel your PRs are ready; then, when reviewers suggest modifications, they may put `awaiting-author` instead, etc, so that it is easy to follow whose turn it is to work on the PR: the author, or the reviewers.)

File: Mathlib/CategoryTheory/Bicategory/LocallyDiscrete.lean [hunk 2]
```diff
@@ -70,64 +90,79 @@ theorem eq_of_hom {X Y : LocallyDiscrete C} {f g : X ⟶ Y} (η : f ⟶ g) : f =
 
 end LocallyDiscrete
 
-variable (C) [Category.{v} C]
+variable {C : Type u} [Category.{v} C]
```
Comment (joelriou) [inline Mathlib/CategoryTheory/Bicategory/LocallyDiscrete.lean:93]: Untested, but it seems to me `C` was declared above.
```suggestion
variable (C)
variable [Category.{v} C]
```

**Comment by joelriou** [comment]
Thanks!

bors d+

File: Mathlib/CategoryTheory/Bicategory/LocallyDiscrete.lean (hunk 2 above)
Comment (callesonne) [reply Mathlib/CategoryTheory/Bicategory/LocallyDiscrete.lean]: Yes thats true! Fixed :)

**Pull request closed without merging**